### PR TITLE
Admin circle page danger zone

### PR DESCRIPTION
**Pull request by sshmm**
## Motivation and Context
implements #966

## Description

1- add danger zone to circle admin page
2- add remove circle modal with remove and cancel buttons
3- on remove circle the circle will be removed and navigation to overview page happens
4- on error show snackbar error

## Test and Deployment Plan

precondition : circle admin
1- Go to circle Admin page and click remove circle button in danger zone
2a - click remove circle button in remove circle modal
Expected:
Navigation to overview page and the circle should not appear from it
2b- click cancel button
Expected:
the modal should be closed
2c- hardcode a wrong circle number and click remove
Expected:
Snackbar should appear with error

## Screenshots (if appropriate):
Screenshot from 2022-08-11 16-05-05

Screenshot from 2022-08-11 16-05-30

### CODE REVIEW

**Comment by crabsinger** [comment]
@sshmm can you update this branch off main and resolve the conflict, then I'll finalize review and merge.

**Comment by sshmm** [comment]
> @sshmm can you update this branch off main and resolve the conflict, then I'll finalize review and merge.

done.

**Review by crabsinger — DISMISSED** [review]
lgtm

**Review by topocount — CHANGES_REQUESTED** [review]
There are a lot of locations where information is contingent on some circle-level lookup succeeding. We should either

a) add the deleted_at clause to the table permissions in hasura
b) make sure that all queries contain this clause.

a is simpler to implement and easier to maintain.

@sshmm reach out to me if you're not sure how to do a. Here are the docs for it: https://hasura.io/docs/latest/auth/authorization/permission-rules/#row-level-permissions

**Comment by sshmm** [comment]
@crabsinger could you merge https://github.com/sshmm/coordinape/tree/AdminCirclePage_DangerZone to this branch because I don't have access to this branch after you rebased it

File: hasura/metadata/databases/default/tables/public_circles.yaml [hunk 1]
```diff
@@ -137,15 +137,16 @@ select_permissions:
...
+      _and:
+      - deleted_at:
+          _is_null: true
+      - users:
+          _and:
+          - profile:
+              id:
+                _eq: X-Hasura-User-Id
+          - deleted_at:
+              _is_null: true
```
Comment (topocount) [inline hasura/metadata/databases/default/tables/public_circles.yaml:149]: This actually breaks functionality for user-level members and fails to return circles a user is not a member in but is within the org. These entries from our test seed are missing in the UI: 
image

I recommend this instead:

```suggestion
      - organization:
          circles:
            users:
              _and:
              - profile:
                  id:
                    _eq: X-Hasura-User-Id
              - deleted_at:
                  _is_null: true
      - deleted_at:
          _is_null: true

```

**Review by topocount — APPROVED** [review]
looks good! passes the test suite locally for me so I'll go ahead and merge it

**Pull request merged**